### PR TITLE
[dist] bring module into 2019

### PR DESCRIPTION
**Pull request by Swaagie**

### CODE REVIEW

**Review by 3rd-Eden — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -23,20 +23,20 @@
...
-    "babel-preset-es2015": "^6.5.0",
-    "babel-register": "^6.26.0",
-    "babelify": "^7.3.0",
-    "istanbul": "~0.4.2",
+    "assume": "^2.1.0",
+    "@babel/cli": "^7.0.0",
+    "@babel/core": "^7.0.0",
+    "@babel/preset-env": "^7.0.0",
+    "@babel/register": "^7.0.0",
+    "babelify": "^10.0.0",
```
Comment (SuzHarrison) [inline package.json]: Do we still need babelify?

**Review by SuzHarrison — APPROVED** [review]
Just one question - otherwise great!

File: .npmrc [hunk 2]
```diff
@@ -1 +0,0 @@
-registry=https://registry.npmjs.org/
```
Comment (DullReferenceException) [inline .npmrc:1]: This can be useful so we don't accidentally install things using GoDaddy's private registry (which would affect the lockfile).

**Review by DullReferenceException — APPROVED** [review]
Just a comment on why I think having `.npmrc` is useful, FWIW. Also agreed that we probably don't need `babelify`. Other than that, looks good to me.

File: package.json [hunk 3]
```diff
@@ -6,9 +6,9 @@
   "browser": "lib/index.js",
   "module": "index.js",
   "scripts": {
-    "prepublish": "npm run transpile",
-    "transpile": "mkdir -p lib && babel -o ./lib/index.js ./index.js",
-    "test": "mocha --spec reporter --compilers js:babel-register ./test.js"
+    "prepublishOnly": "npm run build",
```
Comment (fabiomcosta) [inline package.json:9]: I'd suggest:
```
"prepare": "npm run build",
"prepublishOnly": "npm run test",
```

See:
https://docs.npmjs.com/misc/scripts#prepublish-and-prepare
And
https://github.com/npm/npm/issues/10074

You can make sure you're getting the right behavior for publish with a `npm publish --dry-run` (you'll see it runs `prepare` as well).

One advantage I've noticed with this is that ppl trying to link this package will also have it built (`prepare` runs, but not `prepublishOnly` when linking). There might be other things.

**Comment by Swaagie** [comment]
I would argue that having Godaddy's private registry in the lockfile is either bad configuration or a lockfile generated at the time it wasn't forwarding to the public registry by default (if possible). I still don't feel this is a strong argument to specify a option that is default for npm.

We might not need babelify, others could still depend on it. https://github.com/babel/babelify is still in active development and anyone could be depending on this. Based on the low downloads per week I'm fine removing it. The dependency changes warrant a major release anyways.

**Comment by DullReferenceException** [comment]
The default GoDaddy setup is to have:

```
registry=<our private registry URL>
```

...in your `~/.npmrc`, which is the company-wide recommendation. This causes the `resolved` field in the lockfile to have internal registry links, even if the installed packages come from the npm registry ultimately.

File: package.json (hunk 3 above)
Comment (Swaagie) [reply package.json:9]: Yeah I'm well aware of the details of those different commands, but I also disagree. There is no need to transpile files again on install if those transpiled files are already bundled. It just increases install time for no actual value. If you need to run it again when running `npm link` it's probably better to have a separate `build:watch` running that rebuilds on each change.

I've never been a fan of `prepublish` running on install in the first place. It think I've actually opened an issue about it way back when. But the npm team couldn't get rid of it for legacy reasons. 

As for tests running on `prepublishOnly`, in general it's probably better to have something like Travis in place to run tests on each PR/merge.

**Pull request merged**